forem/forem
# Auto generate profile image for email signups

## What type of PR is this? (check all applicable)

## Description
This PR allows users to sign up via email without uploading a profile image. We generate a default profile image for them.

Also adds some front end bits displaying the required fields for the sign up form.

## Related Tickets & Documents
forem/rfcs#46

## QA Instructions, Screenshots, Recordings
1. Enable email sign up in /admin/customization/config
2. Log out
3. Visit users/sign_up?state=email_signup
4. Sign up as a new user without uploading a profile image
5. Check that the required fields look okay
6. Done!
7. 
![Screenshot_2021-05-03 Welcome - DEV (local)](https://user-images.githubusercontent.com/17884966/116916185-fc7b5380-ac1a-11eb-9e78-a4735994f301.png)

### UI accessibility concerns?
Updated the HTML to make it more readable, and also to skip the asterisk by putting the aria label in the `label`!

## Added tests?


## [Forem core team only] How will this change be communicated?


## [optional] Are there any post deployment tasks we need to perform?
Nope

## [optional] What gif best describes this PR or how it makes you feel?

![Tick tock goes the clock](https://media.giphy.com/media/gf5lHEzYVU8DoVVlhs/giphy.gif)
